src/structlog/_config.py

```diff
@@ -125,14 +127,14 @@ def get_logger(*args, **initial_values):
 
 
 def wrap_logger(
-    logger,
-    processors=None,
-    wrapper_class=None,
-    context_class=None,
-    cache_logger_on_first_use=None,
-    logger_factory_args=None,
-    **initial_values
-):
+    logger: WrappedLogger,
+    processors: Optional[List[Processor]] = None,
+    wrapper_class: Optional[Type[BindableLogger]] = None,
+    context_class: Optional[Type[Context]] = None,
+    cache_logger_on_first_use: bool = None,
+    logger_factory_args: Optional[Tuple[Any, ...]] = None,
+    **initial_values: Any
+) -> Any:
```

Comment:
There are several public functions with arguments of type List or Tuple, like in this case, but code rarely depends on exact type of collection, so it's better to provide user with more generic interface like `Iterable` or `Sequence`, https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html#standard-duck-types

Reply:
Yeah I have kept them to what they used to be in the config for now, but I'm considering to tighten them further.

Reply:
(actually, that weren't that many! Fixed in 3a81ab5ea4b40d5e83fd96c9832ac28c2127245c)

Reply:
Not sure maybe you plan to fix `List` in separate commit, but they should be changed to Iterable or Sequence too. Like processors arg here, user can't pass them in tuple right now.

Reply:
Yeah I'll go through it. However IIRC some rely on being actually lists.

Reply:
I believe I got them in d3128e7